illumos-utils/src/link.rs

```diff
@@ -124,28 +155,24 @@ pub struct InvalidLinkKind(String);
 /// Note that the "ownership" of the VNIC is based on convention;
 /// another process in the global zone could also modify / destroy
 /// the VNIC while this object is alive.
-#[derive(Debug)]
 pub struct Link {
     name: String,
     deleted: bool,
     kind: LinkKind,
+    destructor: Option<Destructor<LinkDestruction>>,
 }
 
-impl Link {
-    /// Takes ownership of an existing VNIC.
-    pub fn wrap_existing<S: AsRef<str>>(
-        name: S,
-    ) -> Result<Self, InvalidLinkKind> {
-        match LinkKind::from_name(name.as_ref()) {
-            Some(kind) => Ok(Self {
-                name: name.as_ref().to_owned(),
-                deleted: false,
-                kind,
-            }),
-            None => Err(InvalidLinkKind(name.as_ref().to_owned())),
-        }
+impl std::fmt::Debug for Link {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(
```

Comment:
Nit: I tend to use [`Formatter::debug_struct`](https://doc.rust-lang.org/std/fmt/struct.Formatter.html#method.debug_struct) when I want to just limit the debug output to a subset of fields.

Reply:
Oooo, TIL - thanks! much better than manually making a string through `write`.